exercises/tweet.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+import sys
+import redis
+import time
+import json
+
+HOST = "redis-13418.c8.us-east-1-4.ec2.cloud.redislabs.com"
+PORT = 13418
+PASS = "QCqZr7caZxWcUahORl0ReOg2JvqD8HZ4"
+POST_PREFIX = "posts:"
+FOLLOWEES_PREFIX = "followees:"
+USER = 'user'
+MESSAGE = 'message'
+TIME = 'time'
+
+
+def post(user: str, message: str, conn):
+    post_to_add = {MESSAGE: message, TIME: time.time()}
+
+    list_name = POST_PREFIX + user
+    message = json.dumps(post_to_add)
+    conn.rpush(list_name, message)
+    result = json.loads(conn.lindex(list_name, -1))
+
+    print(f">{user}:{result.get(MESSAGE)}")
+
+
+def print_timeline(user: str, conn):
+    # Get all my posts
+    all_posts = get_posts_by_user(user, conn)
+
+    # Get all my followees posts
+    for followee in get_followees(user, conn):
+        all_posts += get_posts_by_user(followee, conn)
+
+    sorted_posts = sorted(all_posts, key=lambda k: k.get(TIME), reverse=True)
+
+    for post in sorted_posts:
+        print(f"<{post.get(USER)}>: {post.get(MESSAGE)}")
+
+
+def get_followees(user: str, conn) -> list:
+    return conn.smembers(FOLLOWEES_PREFIX + user)
```

Comment:
It would be much better performance-wise to use `SISMEMBER` on the Redis side than to retrieve all members.

Reply:
we have a SET for every user that includes the users that he follows, how can I get all of them without smembers?